repo.go

```diff
@@ -526,6 +534,50 @@ func (r *Repo) Sign(name string) error {
 	return r.local.SetMeta(name, b)
 }
 
+// AppendSignature allows users to append a signature generated with an external tool.
+func (r *Repo) AppendSignature(name string, signature data.Signature) error {
+	role := strings.TrimSuffix(name, ".json")
+	if !verify.ValidRole(role) {
+		return ErrInvalidRole{role}
+	}
+
+	// Check Key ID is in valid for the role.
+	db, err := r.db()
+	if err != nil {
+		return err
+	}
+	roleData := db.GetRole(role)
+	if roleData == nil {
+		return ErrInvalidRole{role}
+	}
+	if !roleData.ValidKey(signature.KeyID) {
```

Comment:
Good idea, but we should also check that the signature is valid for this key for the signed message.

Reply:
done, added test